src/lib.rs

```diff
@@ -170,6 +170,82 @@ impl<'a> Executor<'a> {
         task
     }
 
+    /// Spawns many tasks onto the executor.
+    ///
+    /// As opposed to the [`spawn`] method, this locks the executor's inner task lock once and
+    /// spawns all of the tasks in one go. With large amounts of tasks this can improve
+    /// contention.
+    ///
+    /// For very large numbers of tasks the lock is occasionally dropped and re-acquired to
+    /// prevent runner thread starvation. It is assumed that the iterator provided does not
+    /// block; blocking iterators can lock up the internal mutex and therefore the entire
+    /// executor.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// use async_executor::Executor;
+    /// use futures_lite::{stream, prelude::*};
+    /// use std::future::ready;
+    ///
+    /// # futures_lite::future::block_on(async {
+    /// let mut ex = Executor::new();
+    ///
+    /// let futures = [
+    ///     ready(1),
+    ///     ready(2),
+    ///     ready(3)
+    /// ];
+    ///
+    /// // Spawn all of the futures onto the executor at once.
+    /// let mut tasks = vec![];
+    /// ex.spawn_many(futures, &mut tasks);
+    ///
+    /// // Await all of them.
+    /// let results = ex.run(async move {
+    ///     stream::iter(tasks).then(|x| x).collect::<Vec<_>>().await
+    /// }).await;
+    /// assert_eq!(results, [1, 2, 3]);
+    /// # });
+    /// ```
+    ///
+    /// [`spawn`]: Executor::spawn
+    pub fn spawn_many<T: Send + 'a, F: Future<Output = T> + Send + 'a>(
+        &self,
+        futures: impl IntoIterator<Item = F>,
+        handles: &mut impl Extend<Task<F::Output>>,
+    ) {
+        let mut active = self.state().active.lock().unwrap();
+
+        for (i, future) in futures.into_iter().enumerate() {
+            // Remove the task from the set of active tasks when the future finishes.
+            let index = active.vacant_entry().key();
+            let state = self.state().clone();
+            let future = async move {
+                let _guard =
+                    CallOnDrop(move || drop(state.active.lock().unwrap().try_remove(index)));
+                future.await
+            };
+
+            // Create the task and register it in the set of active tasks.
+            let (runnable, task) = unsafe {
+                Builder::new()
+                    .propagate_panic(true)
+                    .spawn_unchecked(|()| future, self.schedule())
```

Comment:
Probably out of the scope of this PR, but would there be a reasonable way to add all of the tasks to the injector queue in one go instead of one at a time?

Reply:
it might be good to keep in mind what `spawn_unchecked` does: https://docs.rs/async-task/latest/src/async_task/runnable.rs.html#504-533
Here, it would mean inlining `self.schedule()` basically, which does:
```rust
        move |runnable| {
            state.queue.push(runnable).unwrap();
            state.notify();
        }
```
and we can't really do batch processing in `concurrent-queue` yet, so it would just mean that `state.notify()` gets called once instead of: for every task.

Reply:
Problem is that we still need to do the `notify` anyways in the supplied `schedule` function so that later wake-ups work properly, so doing this properly means first patching in a way for batch processing into `async-task` afaik

Reply:
Yes, right now `concurrent-queue` does not support batch inserts. Making it support batch inserts would require a rewrite of the algorithm, which I would not support.

Reply:
ok, so we can deem this "out of scope because it would require a large-scale rewrite across multiple projects"